mhs/common/mhs_common/workflow/tests/test_synchronous.py

```diff
@@ -0,0 +1,236 @@
+        wd_mock.return_value = wdo
+        self.wf._lookup_to_asid_details = mock.MagicMock()
+        self.wf._lookup_to_asid_details.return_value = test_utilities.awaitable(("yes", "313123"))
+        self.wf._prepare_outbound_message = mock.MagicMock()
+        self.wf._prepare_outbound_message.return_value = test_utilities.awaitable(("123", {"qwe": "qwe"}, "message"))
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
+        future = asyncio.Future()
+        future.set_exception(httpclient.HTTPClientError(code=409))
+        self.transmission.make_request.return_value = future
+
+        error, text, work_description_response = await self.wf.handle_outbound_message(from_asid="202020",
+                                                                                       message_id="123",
+                                                                                       correlation_id="qwe",
+                                                                                       interaction_details={},
+                                                                                       payload="nice message",
+                                                                                       work_description_object=None)
+
+        wdo.set_outbound_status.assert_called_with(work_description.MessageStatus.OUTBOUND_MESSAGE_TRANSMISSION_FAILED)
+        self.assertEqual(error, 500)
+        self.assertEqual(text, 'Error(s) received from Spine: HTTP 409: Conflict')
+
+    @mock.patch('mhs_common.state.work_description.create_new_work_description')
+    @async_test
+    async def test_send_message_http_status_code_err(self, wd_mock):
+        self._setup_success_workflow()
+        wdo = mock.MagicMock()
+        wdo.publish.return_value = test_utilities.awaitable(None)
+        wd_mock.return_value = wdo
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
+        response = mock.MagicMock()
+        response.code = 400
+        response.body = b'err response body'
+        self.transmission.make_request.return_value = test_utilities.awaitable(response)
+
+        error, text, work_description_response = await self.wf.handle_outbound_message(from_asid="202020",
+                                                                                       message_id="123",
+                                                                                       correlation_id="qwe",
+                                                                                       interaction_details={},
+                                                                                       payload="nice message",
+                                                                                       work_description_object=None)
+
+        wdo.set_outbound_status.assert_called_with(work_description.MessageStatus.OUTBOUND_MESSAGE_RESPONSE_RECEIVED)
```

Comment:
Is this the correct status when an error is returned from Spine? The story says it's for when a 200 HTTP status is returned. I think this should be `OUTBOUND_MESSAGE_TRANSMISSION_FAILED`.

I think this test is passing because you need your transmission mock to be returning an `HTTPClientError` with the response code and body set, so we fall into the block that calls the SOAP error handling utility. As it is, you're treating this as a successful response, but with an HTTP error code. I don't think the Tornado HTTP client should do that.